src/lib/comms/sol-oic-server.c

```diff
@@ -189,6 +189,28 @@ static const struct sol_oic_resource_type oic_p_resource_type = {
 };
 
 static CborError
+encode_array_from_bsv(CborEncoder *map, const char *val)
+{
+    const char *p, *last;
+    CborEncoder array;
+    CborError err;
+
+    err = cbor_encoder_create_array(map, &array, CborIndefiniteLength);
+
+    last = val + strlen(val);
+    while (val < last) {
```

Comment:
@cabelitos added a nice `sol_str_slice_split_iterate()` that helps simplifying this kind of loops, consider using it.